etl/steps/data/garden/happiness/2024-06-09/happiness.py

```diff
@@ -0,0 +1,87 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+import owid.catalog.processing as pr
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+REGIONS = {reg: reg_dict for reg, reg_dict in geo.REGIONS.items() if reg != "European Union (27)"}
+REGIONS.update({"World": {}})
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset, previous years and population data.
+    ds_meadow = paths.load_dataset("happiness", version="2024-06-09")
+    ds_prev_years = paths.load_dataset("happiness", channel="garden", version="2023-03-20")
+    ds_population = paths.load_dataset("population", channel="garden")
+
+    # Load regions dataset.
+    ds_regions = paths.load_dataset("regions")
+
+    # Load income groups dataset.
+    ds_income_groups = paths.load_dataset("income_groups")
+
+    # Read table datasets.
+    tb_this_year = ds_meadow["happiness"].reset_index()
+    tb_prev_years = ds_prev_years["happiness"]
+
+    # combine meadow data with previous years
+    tb_this_year["cantril_ladder_score"] = tb_this_year["ladder_score"]
+    cols_overlap = ["country", "cantril_ladder_score", "year"]
+    tb = pr.concat([tb_this_year[cols_overlap], tb_prev_years], ignore_index=True)
+
+    # Harmonize country names
+    tb = geo.harmonize_countries(
+        df=tb, countries_file=paths.country_mapping_path, excluded_countries_file=paths.excluded_countries_path
+    )
+
+    # Process data (add population weighted averages for continents & income groups)
+
+    # save data of Northern Cyrpus and Somaliland to concat later (they do not have population in population dataset)
+    tb_countries_wo_population = tb[tb["country"].isin(["Northern Cyprus", "Somaliland"])]
```

Comment:
You could consider excluding these countries, but I'm also happy if you want to keep them in.

Reply:
Also perhaps a slightly neater way to achieve the same thing would be:

```
msk = tb["country"].isin(["Northern Cyprus", "Somaliland"])
tb_countries_wo_population = tb[msk]
tb = tb[~msk]
```

Reply:
I would keep them in - they only show up if they get selected in the line chart and there I think it might be nice to have them available.